src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md

```diff
@@ -2,43 +2,143 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+Provides dataflow components that are collectively referred to as the *TPL Dataflow Library*.
+This dataflow model promotes actor-based programming by providing in-process message passing for coarse-grained dataflow and pipelining tasks.
 
+Note: This package is part of the larger *Task Parallel library* and is designed to help increase the robustness of concurrency-enabled applications.
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Foundation for message passing and parallelizing CPU-intensive and I/O-intensive applications that have high throughput and low latency.
+* Provides multiple block types for various dataflow operations (e.g., `BufferBlock`, `ActionBlock`, `TransformBlock`).
+* Blocks support linking to form *networks*, allowing for the creation of complex processing topologies.
 
 ## How to Use
 
 <!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
 
+This sample demonstrates a dataflow pipeline that downloads the book "The Iliad of Homer" from a website and searches the text to match individual words with words that reverse the first word's characters.
+
+```csharp
+using System.Net;
+using System.Threading.Tasks.Dataflow;
+
+// Setup blocks
+
+// Downloads the requested resource as a string.
+TransformBlock<string, string> downloadString = new TransformBlock<string, string>(async uri =>
+{
+    Console.WriteLine("Downloading '{0}'...", uri);
+
+    return await new HttpClient(new HttpClientHandler { AutomaticDecompression = DecompressionMethods.GZip }).GetStringAsync(uri);
+});
+
+// Separates the specified text into an array of words.
+TransformBlock<string, string[]> createWordList = new TransformBlock<string, string[]>(text =>
+{
+    Console.WriteLine("Creating word list...");
+
+    // Remove common punctuation by replacing all non-letter characters with a space character.
+    char[] tokens = text.Select(c => char.IsLetter(c) ? c : ' ').ToArray();
+    text = new string(tokens);
+
+    // Separate the text into an array of words.
+    return text.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+});
+
+// Removes short words and duplicates.
+TransformBlock<string[], string[]> filterWordList = new TransformBlock<string[], string[]>(words =>
+{
+    Console.WriteLine("Filtering word list...");
+
+    return words
+       .Where(word => word.Length > 3)
+       .Distinct()
```

Comment:
Why use Distinct when findReversedWords is itself going to use a HashSet? findReversedWords can just iterate through wordsSet rather than iterating through words.